### PR TITLE
State cleanup after booking

### PR DESCRIPTION
**Pull request by mic4ael**

### CODE REVIEW

File: indico/modules/rb_new/client/js/modules/calendar/reducers.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ import {combineReducers} from 'redux';
 
 import {requestReducer} from 'indico/utils/redux';
 import {serializeDate} from 'indico/utils/date';
+import * as bookRoomActions from '../../modules/bookRoom/actions';
```
Comment (ThiefMaster) [inline indico/modules/rb_new/client/js/modules/calendar/reducers.js]: see my other comment

File: indico/modules/rb_new/client/js/modules/calendar/reducers.js [hunk 2]
```diff
@@ -39,6 +40,23 @@ export default combineReducers({
                 return {...state, rows: [], date: action.date};
             case calendarActions.ROWS_RECEIVED:
                 return {...state, rows: action.data};
+            case bookRoomActions.CREATE_BOOKING_SUCCESS: {
+                const {data: {occurrences, room_id: roomId}} = action;
+                const {rows} = state;
+                let newRows = [...rows];
```
Comment (ThiefMaster) [inline indico/modules/rb_new/client/js/modules/calendar/reducers.js]: you overwrite this two lines below, so you can just do `const newRows = rows.map(...);`

**Comment by mic4ael** [comment]
Updated

File: indico/modules/rb_new/client/js/modules/calendar/reducers.js [hunk 3]
```diff
@@ -39,6 +40,21 @@ export default combineReducers({
                 return {...state, rows: [], date: action.date};
             case calendarActions.ROWS_RECEIVED:
                 return {...state, rows: action.data};
+            case bookRoomActions.CREATE_BOOKING_SUCCESS: {
+                const {data: {occurrences, room_id: roomId}} = action;
+                const {rows} = state;
+                const newRows = rows.map((row) => {
+                    const newRow = {...row};
+                    if (row.room_id === roomId) {
```
Comment (ThiefMaster) [inline indico/modules/rb_new/client/js/modules/calendar/reducers.js]:
```
if (row.room_id !== roomId) {
    return row;
}
```

no need to clone a row you don't modify

**Pull request merged**